_references/component-descriptor-structure.md

```diff
@@ -0,0 +1,319 @@
+
+| Property Name | Type     | Required | Description |
+| :------------ | :------: | :------: | :---------- |
+| fields      | `object` | Yes | Object used to define input fields to gather authentication related data from the user. The keys of this object define field names, which must be unique. The values are definitions of the fields. |
+| [oauth1](#oauth1) | `object` |  | Specifies the details about OAuth v1.0 resources. Only used if a `OAuthFieldView` field is defined. |
+| [oauth2](#oauth2) | `object` |  | Specifies the details about OAuth v2.0 resources. Only used if a `OAuthFieldView` field is defined. |
+
+Let's explore an example from the [Petstore API (Node.js)](https://github.com/elasticio/petstore-component-nodejs/blob/master/component.json)
+component. The client authenticated with the Petstore API using an API key
+which is send via an HTTP header. That's why it is sufficient to define in
+the `credentials` object a single field to gather the API key from the user,
+as shown below.
+
+```json
+"credentials": {
+    "fields": {
+      "apiKey": {
+        "label": "API key",
+        "required": true,
+        "viewClass": "TextFieldWithNoteView",
+        "note": "Please use <b>elasticio</b> as API key"
+      }
+    }
+  }
+```
+
+In the example above the `apiKey` field is used to gather user's API key.
+The field is presented in the UI as a simple input field where a user must
+enter his API key. The value of this field will be provided to the component
+at runtime.
+
+## OAuth1
+
+Specifies OAuth 1.0 specific details of the API used by the component. For more
+details about Authenticating with OAuth 1.0 refer to the
+[OAuth 1.0 specification](http://oauth.net/core/1.0/) standard.
+
+| Property Name | Type     | Required | Description |
+| :------------ | :------: | :------: | :---------- |
+| consumer_key | `string` | Yes     | The Consumer Key |
+| consumer_secret | `string` | Yes  | The Consumer Secret |
+| request_token_uri | `string` | Yes | URI to obtain a Request Token |
+| auth_uri | `string` | Yes | URI to obtain User Authorization |
+| access_token_uri | `string` |Yes | URI to obtain an Access Token |
+
+Here is an example implementation of OAuth1 in the Credentials Object:
+
+```json
+{
+"credentials" : {
+   "fields": {
+      "oauth": {
+         "viewClass":"OAuthFieldView",
+         "label":"Twitter Account",
+         "required" : true
+      }
+   },
+   "oauth1":{
+      "consumer_key":"{{TWITTER_API_KEY}}",
+      "consumer_secret":"{{TWITTER_API_SECRET}}",
+      "request_token_uri":"https://api.twitter.com/oauth/request_token",
+      "auth_uri":"https://api.twitter.com/oauth/authorize",
+      "access_token_uri":"https://api.twitter.com/oauth/access_token"
+   }
+  }
+}
+```
+In case of OAuth1 Authentication we must use `fields` and `oauth1` objects together.
+The `fields` object defines the type of input field and the `oauth1` object provides
+the configuration.
```

Comment:
configuration. Please note that the `viewClass` property of the field must be set to `OAuthFieldView`.